### PR TITLE
 allow onyxia to inject OIDC token and refreshtoken in helm chart only on personal project

### LINKED ISSUE #410: Allow oidc  jwt injection in the helm charts
We should allow oidc token injection in the helm charts if values.schema.json specify it to propagate the identity of the user in the container launched.

In order to do that we need to add in OnyxiaValues https://github.com/InseeFrLab/onyxia-web/blob/2bef7775601f421b9fa4be0512ad4310cc780e0c/src/core/ports/OnyxiaApiClient.ts#L154-L210

maybe a new key:

```
 oidc: { 
        token: string; 
        refreshToken: string; 
     }
```

we could then use it in values.schema.json with {{oidc.token}} and {{oidc.refreshToken }}
To do that we need to retrieve this values probably here https://github.com/InseeFrLab/onyxia-web/blob/2bef7775601f421b9fa4be0512ad4310cc780e0c/src/core/usecases/launcher.ts#L984

### PR DESCRIPTION
**Pull request by alexisdondon**
That could be used to propagate identity of user in helm charts 
should fix https://github.com/InseeFrLab/onyxia-web/issues/410

### CODE REVIEW

**Comment by alexisdondon** [comment]
Discuted with Frederic.

We could auhtoriez the ui to inject jwt and refreshtoken to be used in the helm chart but only on personal project to mitigate security issue in group namespaces.

**Comment by alexisdondon** [comment]
what's needed on this to be accepted?

File: src/core/usecases/launcher.ts [hunk 1]
```diff
@@ -1004,6 +1007,11 @@ export const thunks = {
                     };
                 })(),
                 "kaggleApiToken": userConfigs.kaggleApiToken ?? undefined,
+                "oidc": {
+                    "enabled": project.group ? false : true,
+                    "accessToken": project.group ? undefined : oidcClient.accessToken,
+                    "refreshToken": project.group ? undefined : oidcClient.refreshToken
```
Comment (garronej) [inline src/core/usecases/launcher.ts:1013]: The type system should be more transparent here.  
`project.group` should be of type `string | undefined`.  
Also we introduce the notion of `group` for only using it to tell if it's an empty string or not.  
We should instead have a `project.isPersonal: boolean`.  
Also, until now we assumed that the first project was the personal project. So we should update the part of the code where we made this assemption (serach for `projects[0]`)

File: src/core/adapters/phonyOidcClient.ts [hunk 2]
```diff
@@ -51,7 +51,7 @@ export function createPhonyOidcClient(params: {
                 "xxx"
             );
 
-            return { accessToken };
+            return { accessToken, refreshToken: "phonyToken" };
```
Comment (garronej) [inline src/core/adapters/phonyOidcClient.ts]: "refreshToken": "phonyToken"

File: src/core/adapters/officialOnyxiaApiClient.ts [hunk 3]
```diff
@@ -701,6 +701,7 @@ export function createOfficialOnyxiaApiClient(params: {
                         projects: {
                             id: string;
                             name: string;
+                            group: string;
```
Comment (garronej) [inline src/core/adapters/officialOnyxiaApiClient.ts:704]: Are we sure of that?   
Is it not `group?: string` ?  
Seems more likely.

File: src/core/adapters/keycloakOidcClient.ts [hunk 4]
```diff
@@ -72,6 +72,7 @@ export async function createKeycloakOidcClient(params: {
     const oidcClient = id<OidcClient.LoggedIn>({
         "isUserLoggedIn": true,
         "accessToken": keycloakInstance.token!,
+        "refreshToken": keycloakInstance.refreshToken!,
```
Comment (garronej) [inline src/core/adapters/keycloakOidcClient.ts]: I'm pretty sure that when the accessToken is refreshed, the refreshToken changes as well.  
Also, you should pull, this have been updated. It's now `getToken()`

**Comment by garronej** [comment]
Hey @alexisdondon,  

Thank you for your work. Here is my review, sorry for the delay.

**Comment by garronej** [comment]
UP?

**Pull request closed without merging**